pkg/node/address_linux.go

```diff
@@ -31,6 +32,15 @@ import (
 	"github.com/vishvananda/netlink"
 )
 
+type labeledIP struct {
+	IP    net.IP
+	Label string
+}
+
+func (ip labeledIP) IsSecondary() bool {
+	return strings.Contains(ip.Label, ":")
```

Comment:
Is this preference order described somewhere else than https://www.ietf.org/rfc/rfc3484.txt? That RFC seems to indicate the algorithm to take labels into account is actually a bit more complex and involves the destination address' label.

Reply:
I don't know of any RFC describing this. I just remember this as a convention on Linux systems.